File: src/main/java/com/hoi4utils/clauzewitz/data/country/Character.java
package main.java.com.hoi4utils.clauzewitz.data.country;

import main.java.com.hoi4utils.clauzewitz.localization.Localizable;
import org.jetbrains.annotations.NotNull;

import java.util.Collection;
import java.util.List;
import java.util.Map;

public class Character implements Localizable {
    String id;

    @Override
    public @NotNull Map<Property, String> getLocalizableProperties() {
        return Map.of(Property.NAME, id);
    }

    @Override
    public @NotNull Collection<? extends Localizable> getLocalizableGroup() {
        return List.of(this); // todo improve
    }
}

File: _src_old/main/scala/com/hoi4utils/hoi4mod/history/countries/Character.scala
package com.hoi4utils.hoi4mod.history.countries

import com.hoi4utils.hoi4mod.localization.{Localizable, Property}

import scala.collection.mutable

class Character extends Localizable {
  private val id: String = null

  override def getLocalizableProperties: mutable.Map[Property, String] = mutable.Map(Property.NAME -> id)

  override def getLocalizableGroup: Iterable[? <: Localizable] = List(this) // todo improve
}

